UnityProjectDP/Assets/Scripts/Visualization/UI/PopUps/AbstractPopUp.cs

```diff
@@ -1,26 +1,40 @@
-﻿using TMPro;
+﻿using AnimArch.Extensions;
+using AnimArch.Visualization.Diagrams;
 using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
 
 namespace AnimArch.Visualization.UI
 {
     public abstract class AbstractPopUp : MonoBehaviour
     {
         public GameObject panel;
-        public TMP_InputField inp;
-        protected TMP_Text className;
 
-        public virtual void ActivateCreation(TMP_Text classTxt)
+        protected void ChangeStateOfButtons(bool state)
         {
+            DiagramPool.Instance.ClassDiagram.graph.GetComponentsInChildren<GraphicRaycaster>()
+                .ForEach(x => x.enabled = state);
+            var parent = transform.parent.parent;
+            parent.Find("RightMenu").GetComponentsInChildren<Button>()
+                .ForEach(x => x.enabled = state);
+            parent.Find("TopMenu").GetComponentsInChildren<Button>()
+                .ForEach(x => x.enabled = state);
+            parent.Find("TopMenu").GetComponentsInChildren<EventTrigger>()
+                .ForEach(x => x.enabled = state);
+        }
+
+        public virtual void ActivateCreation()
+        {
+            ChangeStateOfButtons(false);
```

Comment:
mozno by dokonca bolo este kusok lepsie pre citatelnost, spravit `DisableButtons()` `EnableButtons()` resp. ak sa to tyka presne nejakych buttonov tak `DisableXXButtons`, kde XX ked sa to tyka napr. menu bolo `DisableMenuButtons`, ktore by v sebe interne volali s true resp false terajsi `ChangeStateOfButtons`. Ak by sme sa s tou metodou vyhrali, tak by sme to mozno vedeli aj zovseobecnit na akykolvek button co pouzivame? Pripadne ak to nechame takto, tak podla vzoru o riadok nizssie `SetButtonsActive(bool active)`